### PR TITLE
refactor(es6): Storage: use es6 class

### PR DESCRIPTION
**Pull request by jkwlui**
Part of #173

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -85,28 +84,377 @@ const File = require('./file.js');
...
+
+    super(config, options);
+
+    /**
+     * Reference to {@link Storage.acl}.
+     *
+     * @name Storage#acl
+     * @see Storage.acl
+     */
+    this.acl = Storage.acl;
```
Comment (kinwa91) [inline src/index.js]: Object properties will be defined in the class directly (not from constructor) when we convert to Typescript.

File: package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "license": "Apache-2.0",
   "author": "Google Inc.",
   "engines": {
-    "node": ">=4"
+    "node": ">=6"
```
[the remark this answers is not in the document]
Comment (kinwa91) [reply package.json]: Failure of node4 build is expected.

[the remark this answers is not in the document]
Comment (crwilcox) [reply package.json]: Can you fix the circle ci file to not run node4? @JustinBeckwith did this for another repo I believe already so you may be able to just copy that one.

[the remark this answers is not in the document]
Comment (crwilcox) [reply package.json]: https://github.com/googleapis/nodejs-spanner/pull/226/files

[the remark this answers is not in the document]
Comment (JustinBeckwith) [reply package.json]: Yup! Please drop node.js 4 support in a different PR.  Make sure to update package.json, and the CircleCI config :)

**Review by JustinBeckwith — APPROVED** [review]

**Comment by JustinBeckwith** [comment]
Just need to make sure #242 lands first :)

**Pull request merged**